examples/ascii.rs

```diff
@@ -7,19 +7,20 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     // Open the port, home device 1, and wait for it to finish.
     let mut port = Port::open_serial("/dev/ttyUSB0")?;
-    port.command_reply_with_check(
-        (1, "home"),
+    port.command_reply((1, "home"))?.flag_ok_and(
         check::warning_in(("WR", "WH", Warning::NONE)), // Ignore warnings about it being unhomed.
     )?;
-    port.poll_until_idle(1)?;
+    port.poll_until_idle(1, check::flag_ok())?;
 
     // Move towards the end of travel and monitor position as it goes.
     // Once the position exceeds a specific limit, interrupt the motion.
-    port.command_reply((1, 1, "move max"))?;
-    port.poll_until((1, 1, "get pos"), |reply| {
+    port.command_reply((1, 1, "move max"))?
+        .flag_ok_and(check::warning_is_none())?;
+    port.poll_until((1, 1, "get pos"), check::flag_ok(), |reply| {
         let pos: i32 = reply.data().parse().unwrap();
         pos >= 100000
     })?;
-    port.command_reply_with_check((1, "stop"), check::warning_is("NI"))?;
+    port.command_reply((1, "stop"))?
+        .flag_ok_and(check::warning_is("NI"))?;
```

Comment:
Code ends up with _a lot_ of `flag_ok`s. It seems like 99% of the time, the flag should be `ok`. Can we make that the default some how so the common case is clearer, even if it makes the odd case of wanting an `RJ` more verbose? One option would be to check the reply flag when we validate the header. If we did that, the error probably shouldn't be called `CheckFlagError` anymore, since it isn't a `Check` failure anymore.

Reply:
Given that the Chain layer has been/is bring added, perhaps the port layer should be lower level, requiring explicit validation of each message, while the higher level APIs have a standard checking protocol (must always be OK, no F-level warnings, perhaps not even some W-level warnings like WR or WH for motion routines.

Reply:
This is still better than what was there before. You could rarely every use the methods without `with_check` because there was almost always some warning flag to consider and the default check was `strict`. This is more compact and reads more easily. Higher level APIs on the Chain could reduce that if we want, but that is for a separate MR.